pkg/providers/azure/README.md

```diff
@@ -3,69 +3,85 @@ The Azure Key Vault Provider offers two modes for accessing a Key Vault instance
-        array:                        # array of objects
-          - |
-            objectName: secret1
-            objectType: secret        # object types: secret, key or cert
-            objectVersion: ""         # [OPTIONAL] object versions, default to latest if empty
-          - |
-            objectName: key1
-            objectType: key
-            objectVersion: ""
-      resourceGroup: ""               # resource group of the KeyVault
-      subscriptionId: ""              # subscription ID of the KeyVault
-      tenantId: ""                    # tenant ID of the KeyVault
-      ...
-```
+1. Add your service principal credentials as a Kubernetes secrets accessible by the Secrets Store CSI driver.
+
+    ```bash
+    kubectl create secret generic secrets-store-creds --from-literal clientid=<CLIENTID> --from-literal clientsecret=<CLIENTSECRET>
+    ```
+
+    Ensure this service principal has all the required permissions to access content in your Azure key vault instance.
+    If not, you can run the following using the Azure cli:
+
+    ```bash
+    # Assign Reader Role to the service principal for your keyvault
+    az role assignment create --role Reader --assignee <principalid> --scope /subscriptions/<subscriptionid>/resourcegroups/<resourcegroup>/providers/Microsoft.KeyVault/vaults/<keyvaultname>
+
+    az keyvault set-policy -n $KV_NAME --key-permissions get --spn <YOUR SPN CLIENT ID>
+    az keyvault set-policy -n $KV_NAME --secret-permissions get --spn <YOUR SPN CLIENT ID>
+    az keyvault set-policy -n $KV_NAME --certificate-permissions get --spn <YOUR SPN CLIENT ID>
+    ```
+
+1. Update [this sample deployment](examples/v1alpha1_secretproviderclass.yaml) to create a `secretproviderclasses` resource to provide Azure-specific parameters for the Secrets Store CSI driver.
+
+    ```yaml
+    apiVersion: secrets-store.csi.k8s.com/v1alpha1
+    kind: SecretProviderClass
+    metadata:
+      name: azure-kvname
+    spec:
+      provider: azure                   # accepted provider options: azure or vault
+      parameters:
+        usePodIdentity: "false"         # [OPTIONAL for Azure] if not provided, will default to "false"
+        keyvaultName: "kvname"          # the name of the KeyVault
+        objects:  |
+          array:
+            - |
+              objectName: secret1
+              objectType: secret        # object types: secret, key or cert
+              objectVersion: ""         # [OPTIONAL] object versions, default to latest if empty
+            - |
+              objectName: key1
+              objectType: key
+              objectVersion: ""
+        resourceGroup: "rg1"            # the resource group of the KeyVault
+        subscriptionId: "subid"         # the subscription ID of the KeyVault
+        tenantId: "tid"                 # the tenant ID of the KeyVault
+
+    ```
+
+    |Name|Required|Description|Default Value|
+    |---|---|---|---|
+    |providerName|yes|specify name of the provider|""|
```

Comment:
nit: just so we use the same key name as the keys in yaml. %s/providerName/provider